ui/src/app/base/components/FabricField/FabricField.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from "react";
+
+import { Select, Spinner } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import type { Props as FormikFieldProps } from "app/base/components/FormikField/FormikField";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+
+type Props = Partial<FormikFieldProps>;
+
+export const FabricField = ({ ...props }: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const fabrics = useSelector(fabricSelectors.all);
+  const fabricsLoaded = useSelector(fabricSelectors.loaded);
+
+  useEffect(() => {
+    dispatch(fabricActions.fetch());
+  }, [dispatch]);
+
+  if (!fabricsLoaded) {
+    return <Spinner />;
+  }
+
+  return (
+    <FormikField
+      component={Select}
+      label="Fabric"
+      name="fabric"
+      options={fabrics.map((fabric) => ({
```

Comment:
In most selects I've included a disabled default option that just says "Select foo" - it was a carry over from the angular version but we should at least keep them consistent

Reply:
OK, I'll add them.